### PR TITLE
Emit unhandledRejection on Internet Explorer (+ fixes on Windows)

### PR DESCRIPTION
**Pull request by Volune**
Hello,

This Pull Request is to emit the `unhandledRejection` event on Internet Explorer, which doesn't support the `CustomEvent` class.

I've manually tested on Internet Explorer 9 and 11.

Also provide some fixes to build and test on Windows.

### CODE REVIEW

File: lib/makePromise.js [hunk 1]
```diff
@@ -895,27 +903,37 @@ define(function() {
...
-						var ev = new CustomEvent(type, {
-							detail: {
-								reason: rejection.value,
-								key: rejection
-							},
-							bubbles: false,
-							cancelable: true
+			}
+			if(typeof self !== 'undefined' && typeof CustomEvent === 'function') {
+				if (hasCustomEvent()) {
```
Comment (briancavalier) [inline lib/makePromise.js]: Can this `if(hasCustomEvent())` can be merged into the enclosing `if`, and then the `if` below can become an `else if`?

Comment (Volune) [inline lib/makePromise.js]: Updated, let me know if you find anything else to improve.

File: lib/makePromise.js [hunk 2]
```diff
@@ -882,6 +882,28 @@ define(function() {
...
+			if(typeof CustomEvent === 'function') {
+				try {
+					var ev = new CustomEvent('unhandledRejection');
+					return ev instanceof CustomEvent;
+				} catch (ignoredException) {}
+			}
+			return false;
+		}
+
+		function hasInternetExplorerCustomEvent() {
```
Comment (briancavalier) [inline lib/makePromise.js]: :+1:

File: lib/makePromise.js [hunk 3]
```diff
@@ -915,7 +931,19 @@ define(function() {
 
 						return !self.dispatchEvent(ev);
 					};
-				}(noop, self, CustomEvent));
+				}(self, CustomEvent));
+			} else if(typeof self !== 'undefined' && hasInternetExplorerCustomEvent()) {
```
Comment (briancavalier) [inline lib/makePromise.js]: :+1:

**Comment by briancavalier** [comment]
Thanks @Volune!

**Pull request merged**